wip/services/exec/store.go

```diff
@@ -145,7 +184,61 @@ func (p postgresStore) UpdateStatus(id string, status types.Status) error {
 	return nil
 }
 
-func dbExecToExec(dbe db.UnweaveExec) types.Exec {
+func addSSHKeyToExec(ctx context.Context, exec types.Exec, keys []db.UnweaveSshKey) error {
+	for _, key := range keys {
+		err := db.Q.ExecSSHKeyInsert(ctx, db.ExecSSHKeyInsertParams{
+			ExecID:   exec.ID,
+			SshKeyID: key.ID,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func createSSHKeys(ctx context.Context, createdByID string, keys []types.SSHKey) error {
+	for _, key := range keys {
+		key := key
+		exists, err := db.Q.SSHKeyGetByPublicKey(ctx, db.SSHKeyGetByPublicKeyParams{
+			PublicKey: *key.PublicKey,
+			OwnerID:   createdByID,
+		})
+		if err != nil && err != sql.ErrNoRows {
+			return err
+		}
+		if exists.PublicKey != "" {
+			continue
+		}
+		err = db.Q.SSHKeyAdd(ctx, db.SSHKeyAddParams{
+			OwnerID:   createdByID,
+			Name:      key.Name,
+			PublicKey: *key.PublicKey,
+		})
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func dbSSHKeyToSSHKey(ks []db.UnweaveSshKey) (res []types.SSHKey) {
+	for _, k := range ks {
+		k := k
+		res = append(res, types.SSHKey{
+			Name:       k.Name,
+			PublicKey:  &k.PublicKey,
+			PrivateKey: nil,
+			CreatedAt:  &k.CreatedAt,
+		})
+	}
```

Comment:
Again, we know the length of the slice we need so it's better to pre-define it.

Reply:
Happy to standardise to defining capacity moving forward